app/handlers/customer/delete.js

```diff
@@ -0,0 +1,150 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+const {
+  DynamoDBClient,
+  BatchWriteItemCommand,
+  ScanCommand,
+  QueryCommand,
+} = require("@aws-sdk/client-dynamodb");
+
+const { unmarshall } = require("@aws-sdk/util-dynamodb");
+
+const ddbClient = new DynamoDBClient({ region: process.env.AWS_REGION });
+
+exports.handler = async function (event, context) {
+  let deleteResp = {};
+  const findCustomerParams = {
+    FilterExpression: "SK = :sk AND cognitoIdentityId = :identifier",
+    ExpressionAttributeValues: {
+      ":sk": { S: "COGNITO_IDENTITY_ID" },
+      ":identifier": { S: event.requestContext.identity.cognitoIdentityId },
+    },
+    TableName: process.env.CUSTOMER_TABLE_NAME,
+  };
+
+  const { Items: idRecords } = await ddbClient.send(
```

Comment:
If we take customerId as input parameter, then it would allow any authenticated user to delete any other user using their customer id. This is the only way that I could find to make sure that the customer is only able to delete their own data. I think we can address this if/when we restructure to implement single table design or separate databases for different micro services.

Reply:
@dmahapatro ^

Reply:
Fair enough.